domain/src/main/java/life/qbic/apps/datamanager/services/UserRegistrationService.java

```diff
@@ -81,7 +95,41 @@ public void handleEvent(UserRegistered event) {
    * @param userId the id of the user to be activated
    * @since 1.0.0
    */
-  public void activateUser(String userId) {
+  public void confirmUserEmail(String userId) {
+    DomainEventPublisher.instance().subscribe(new DomainEventSubscriber<UserActivated>() {
+
+      @Override
+      public Class<UserActivated> subscribedToEventType() {
+        return UserActivated.class;
+      }
+
+      @Override
+      public void handleEvent(UserActivated event) {
+        storeEvent(event);
+        sendNotification(event);
+        //TODO unsubscribe
+      }
+
+      private void sendNotification(UserActivated event) {
+        var notificationId = notificationService.newNotificationId();
+        var notification =
+            Notification.create(
+                UserActivated.class.getSimpleName(),
+                event.occurredOn(),
+                notificationId,
+                event);
+        notificationService.send(notification);
+      }
+
+      private void storeEvent(UserActivated event) {
+        eventStore.append(event);
+      }
+    });
+    UserRepository userRepository = UserRepository.getInstance(userDataStorage);
+    Optional<User> optionalUser = userRepository.findById(userId);
+    optionalUser.ifPresentOrElse(User::confirmEmail, () -> {
+      throw new RuntimeException("Unknown user. Could not confirm the email address.");
+    });
```

Comment:
You need to store the updated user state

Reply:
Yes good catch. I implemented that